### DISCUSSION TITLE
Question: Can rapier2d be used for physics on a toroid/on a 2d surface wrapping in x- and y-directions?

### DISCUSSION THREAD

**Discussion by maghoff** [dimforge/rapier#386] [Q&A]
Some games take place on a toroid: The world is two-dimensional, but it _wraps_ on the edges, such that anything that goes too far right reappears on the left and so on. This topology is that of a torus.

Can rapier2d handle such worlds out of the box? Could I alternatively implement this on top of rapier2d somehow?

**Comment by sebcrozet** [comment]
Hi! No, `rapier` won’t handle non-euclidean spaces automatically.

**Reply by maghoff** [reply]
Thanks. Is there API that can be used to implement the desired functionality? I'm not super well-versed in physics engines, but I imagine it goes in phases: 1. collision detection, 2. apply forces (or something like that). Would it be possible to amend the detected collisions by injecting an extra pass to consider bodies that straddle the boundaries?
